### PR TITLE
Fix parameter binding for table functions 

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Fixes https://github.com/crate/crate/issues/12442

## Checklist

 - [ ] Added an entry in `CHANGES.txt` for user facing changes
 - [ ] Updated documentation & `sql_features` table for user facing changes
 - [ ] Touched code is covered by tests
 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [ ] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/io/crate/integrationtests/PostgresITest.java [hunk 1]
```diff
@@ -1100,6 +1100,30 @@ public void test_metadata_calls_do_not_query_pg_catalog_tables_repeatedly() thro
         }
     }
 
+    @Test
+    public void test_table_function_without_from_can_bind_parameters() throws Exception {
```
Comment (BaurzhanSakhariev) [inline server/src/test/java/io/crate/integrationtests/PostgresITest.java:1104]: Not sure it's needed, created a planner test as well - should I drop it?

Comment (mfussenegger) [reply server/src/test/java/io/crate/integrationtests/PostgresITest.java:1104]: I think planner test would be good enough, but also fine to keep it. No strong opinion

**Review by mfussenegger — APPROVED** [review]
:+1:

File: server/src/test/java/io/crate/integrationtests/PostgresITest.java (hunk 1 above)
Comment (BaurzhanSakhariev) [reply server/src/test/java/io/crate/integrationtests/PostgresITest.java:1104]: will keep it for now

**Pull request merged**